src/app/collective-rewards/rewards/backers/RBI.tsx

```diff
@@ -64,3 +62,52 @@ export const RBI: FC<RBIProps> = ({ backer, tokens, ...metricsCardProps }) => {
     </MetricsCardWithSpinner>
   )
 }
+
+const RBIWStateSync: FC<RBIProps> = ({ backer, tokens, ...metricsCardProps }) => {
+  const {
+    data: backerStakingHistory,
+    isLoading: backerStakingHistoryLoading,
+    error: backerStakingHistoryError,
+  } = useGetBackerStakingHistoryWithStateSync(backer)
+  const {
+    data: rbiPct,
+    isLoading: rbiPctLoading,
+    error: rbiPctError,
+  } = useGetBackerRBI(backerStakingHistory, tokens)
+
+  const isLoading = backerStakingHistoryLoading || rbiPctLoading
+  const error = backerStakingHistoryError ?? rbiPctError
+  useHandleErrors({ error, title: 'Error loading RBI with state sync' })
+
+  return <RBIContent rbiPct={rbiPct} isLoading={isLoading} {...metricsCardProps} />
+}
+
+const RBIWTheGraph: FC<RBIProps> = ({ backer, tokens, ...metricsCardProps }) => {
+  const {
+    data: backerStakingHistory,
+    isLoading: backerStakingHistoryLoading,
+    error: backerStakingHistoryError,
+  } = useGetBackerStakingHistoryWithGraph(backer)
+  const {
+    data: rbiPct,
+    isLoading: rbiPctLoading,
+    error: rbiPctError,
+  } = useGetBackerRBI(backerStakingHistory, tokens)
+
+  const isLoading = backerStakingHistoryLoading || rbiPctLoading
+  const error = backerStakingHistoryError ?? rbiPctError
+  useHandleErrors({ error, title: 'Error loading RBI with the graph' })
+
+  return <RBIContent rbiPct={rbiPct} isLoading={isLoading} {...metricsCardProps} />
+}
+
+interface RBIProps extends MetricsCardProps {
+  backer: Address
+  tokens: Record<string, Token>
+}
+export const RBI: FC<RBIProps> = props => {
+  const {
+    flags: { use_state_sync },
+  } = useFeatureFlags()
+  return use_state_sync ? <RBIWStateSync {...props} /> : <RBIWTheGraph {...props} />
```

Comment:
for me we shouldn't use a flag for this, but an enum (e.g. data_source: [smart_contracts, subgraph, api])

Reply:
Introduced a TODO, to handle multiple data sources.